File: Python/BinarySearch.py
import sys
def binarSearch(arr, key ):	
	lo = 1
	hi = len(arr)-1 
	while True:
		mid = (lo + hi )//2
		if arr[mid] == key: return key 
		if arr[mid] > key :
			print(f"big {arr[mid]}")
			hi = mid-1
		elif arr[mid]<key:
			lo = mid
			print(f"small {arr[mid]}")

arr = [i for i in range(1,1000,2)]
print(binarSearch(arr, 241))

File: CPP/BinarySearch.cpp
#include<iostream>
#include<vector>
//#include<bits/stdc++.h>
using namespace std;

int binarySearch(vector<int> arr, int key){
	if (arr[0] == key )
		return 0;
	int low = 1 , hi = arr.size()-1;
	while(true){
		int mid = (low+hi)/2;
		if (arr[mid]==key)
			return mid;
		if (arr[mid] < key){
			cout<<"too small ";
			low = mid+1;
		}else if (arr[mid]>key){
			cout<<"too high ";
			hi = mid-1;
		}
	}
	return 0;
}

int main(){

	#ifndef ONLINE_JUDGE
		freopen("input.txt","r",stdin);
		freopen("output.txt","w",stdout);
	#endif
	
	vector<int> arr ;
	for(int i = 0 ; i < 2000 ;i+=2)
		arr.push_back(i);

	for(auto i : arr)
		cout<<i<<" ";

	cout<<"\nPresent at index : "<<binarySearch(arr,234);
	return 0;
}
